File: src/_02_StructuralPattern/_02_04_Decorator/CommentDecorator.js
import CommentService from "./CommentService.js";

export default class CommentDecorator extends CommentService {
  /** @type { CommentService } */
  #commentService;

  /** @param { CommentService } */
  constructor(commentService) {
    super();
    this.#commentService = commentService;
  }

  /**
   * @override 
   * @param { string } comment 
   */
  addComment(comment) {
    this.#commentService.addComment(comment);
  }
}

File: src/_02_StructuralPattern/_02_04_Decorator/java/after/CommentDecorator.java
package _02_StructuralPattern._02_04_Decorator.java.after;

public class CommentDecorator implements CommentService {

  private CommentService commentService;
  
  public CommentDecorator(CommentService commentService) {
    this.commentService = commentService;
  }
  
  @Override
  public void addComment(String comment) {
    commentService.addComment(comment);
  }
  
}
